backend/tests/infrastracture/graph/test_gql_retry_backoff.py

```diff
@@ -0,0 +1,34 @@
+import pytest
+from datetime import datetime, timedelta
+
+
+from gql import gql
+from gql.transport.exceptions import TransportQueryError
+
+from app.extensions import gql_factory
+
+
+def test_with_failure(mock_failing_gql):
+    """
+    1. mock gql to throw TransportQueryError
+    2. check if exception is being thrown
+    3. check if retry timeout is in place and call takes more than 2 seconds
+    """
+    query = gql(
+        """
+query {
+  epoches {
+    epoch
+    fromTs
+    toTs
+    duration
+    decisionWindow
+  }
+}
+    """
+    )
+    start = datetime.now()
+    with pytest.raises(TransportQueryError):
+        gql_factory.build().execute(query)
+    finish = datetime.now()
+    assert finish - start > timedelta(seconds=2)
```

Comment:
maybe a better option will be use pytest approx with some given fault toleration instead of using >

Reply:
Timers in give hard guarantee of not returning control sooner. This should be safe.

Reply:
the idea of that was to make sure that the substraction won't be equal, e.g 4 seconds when we have max_time set to 2 seconds, where approx would give us FAILURE but > operator a false positive result

Reply:
wdyt? it may stay as it is but let's be sure about that

Reply:
Timedelta precision is in microseconds. I think this is very unlikely. But yes, there is a chance of us getting a false positive. Instead of "more" we should use "more or equal" here.

Reply:
+1